test/index.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const { mockClient } = require('aws-sdk-client-mock');
+const { ECS, DescribeTaskDefinitionCommand, RunTaskCommand } = require("@aws-sdk/client-ecs");
+const { CloudWatchLogs, GetLogEventsCommand } = require("@aws-sdk/client-cloudwatch-logs");
+const expect = require('expect.js');
+const { promisify } = require('node:util');
+const index = promisify(require('../index'));
+
+describe('index', function () {
+  const cwlMock = mockClient(CloudWatchLogs);
+  const ecsMock = mockClient(ECS);
+  afterEach(() => {
+    cwlMock.reset();
+    ecsMock.reset();
+  });
+
+  it('should do the thing', async function () {
+    const options = {
+      taskDefinitionArn: 'task-definition.arn',
+      containerName: 'meow'
+    };
+
+    ecsMock.on(DescribeTaskDefinitionCommand).callsFake(async params => {
+      expect(params.taskDefinition).to.eql(options.taskDefinitionArn)
+
+      return {
+        taskDefinition: {
+          taskDefinitionArn: options.taskDefinitionArn,
+          containerDefinitions: [{
+            name: options.containerName,
+            logConfiguration: {
+              logDriver: 'awslogs',
+              options: { 'awslogs-group': '', 'awslogs-stream-prefix': '' },
+            }
+          }],
+        }
+      }
+    });
+
+    // thread the randon EOF through to kill the stream
+    let eofSet;
+    const eof = new Promise(r => {
+      eofSet = r;
+    })
+    ecsMock.on(RunTaskCommand).callsFake(async params => {
+      expect(params.taskDefinition).to.eql(options.taskDefinitionArn)
+
+      eofSet(params.overrides.containerOverrides[0].command[2].split(' ')[9])
```

Comment:
I don't really get what this does

Reply:
I'll add a comment, but yeah this is ugliest bit of this. This is the only place we can grab the random value the log stuff uses to detect the end of the task run.